src/importlinter/domain/imports.py

```diff
@@ -86,3 +86,29 @@ def __str__(self) -> str:
 
     def __hash__(self) -> int:
         return hash((str(self), self.line_contents))
+
+
+class ImportExpression(ValueObject):
+    """
+    A user-submitted expression describing an import or set of imports.
+
+    Sets of imports are notated using * wildcards.
+    These wildcards can stand in for a module name or part of a name, but they do
+    not extend to subpackages.
+
+    For example, "mypackage.*" refers to every child subpackage of mypackage.
+    It does not, however, include more distant descendants such as mypackage.foo.bar.
+    """
+
+    def __init__(self, importer: str, imported: str) -> None:
+        self.importer = importer
+        self.imported = imported
```

Comment:
Perhaps we should raise a `ValueError` if we get a malformed import expression, what do you think?

Reply:
I guess, then it make sense to extract the import expression validationto a util function or NOT to validate in ImportExpressionField